app/controllers/licence_controller.rb

```diff
@@ -14,101 +14,64 @@ class LicenceController < ApplicationController
-  LICENCE_SLUGS_WITH_COUNTY_TIER_OVERRIDE = [
-    'scaffolding-and-hoarding-licence',
-    'skip-operator-licence',
-    'permission-to-place-tables-and-chairs-on-the-pavement',
-    'pavement-or-street-display-licence',
-    'petroleum-storage-licence',
-    'weighbridge-operator-certificate',
-    'performing-animals-registration',
-    'approval-of-premises-for-civil-marriage-or-civil-partnership',
-    'licence-projection-over-highway-england-wales',
-  ].freeze
-
-  def search
-    @interaction_details = licence_details
-
-    if request.post?
-      if postcode_search_submitted?
-        @postcode = postcode
-        @location_error = location_error
-
-        if local_authority_slug
-          redirect_to licence_authority_path(slug: params[:slug], authority_slug: local_authority_slug)
-        end
-      end
-    elsif @publication.continuation_link.present?
+  def start
+    if @publication.continuation_link.present?
       render :continues_on
-    elsif authority_choice_submitted?
+    elsif @licence_details.local_authority_specific? && postcode_search_submitted?
+      @postcode = postcode
+      @location_error = location_error
+
+      redirect_to licence_authority_path(slug: params[:slug], authority_slug: local_authority_slug) if local_authority_slug
+    elsif @licence_details.single_licence_authority_present?
+      redirect_to licence_authority_path(slug: params[:slug], authority_slug: @licence_details.authority['slug'])
+    elsif @licence_details.multiple_licence_authorities_present? && authority_choice_submitted?
       redirect_to licence_authority_path(slug: params[:slug], authority_slug: CGI.escape(params[:authority][:slug]))
-    elsif single_authority_interaction_details_present?
-      redirect_to licence_authority_path(slug: params[:slug], authority_slug: @interaction_details[:authority]["slug"])
     end
   end
 
   def authority
     if @publication.continuation_link.present?
       redirect_to licence_path(slug: params[:slug])
-    elsif location_specific_licence?
-      raise RecordNotFound unless artefact_with_snac
-      @publication = PublicationPresenter.new(artefact_with_snac)
-      @interaction_details = licence_details_for_snac(params[:authority_slug], snac_from_slug)
-    else
-      @interaction_details = licence_details(params[:authority_slug])
+    elsif @licence_details.local_authority_specific?
+      @licence_details = LicenceDetailsPresenter.new(licence_details_from_api_for_local_authority, nil, params[:interaction])
     end
   end
 
 private
 
-  def artefact_with_snac
-    return nil if snac_from_slug.blank?
+  def set_publication
+    @publication = LicencePresenter.new(artefact)
+    @licence_details = LicenceDetailsPresenter.new(licence_details_from_api, params["authority_slug"], params[:interaction])
+    set_language_from_publication
+  end
+
+  def licence_details_from_api(snac = nil)
+    return {} if @publication.continuation_link.present?
```

Comment:
We thought about adding a wrapper predicate method but it always just sounded very similar to 'continuation link' so gave up.